packages/tempus/unit_test/Tempus_UnitTest_DIRK.cpp

```diff
@@ -0,0 +1,181 @@
+          }
+
+          break;
+        }
+      case StepperRKAppAction<double>::AFTER_SOLVE:
+      case StepperRKAppAction<double>::BEFORE_EXPLICIT_EVAL:
+      case StepperRKAppAction<double>::END_STAGE:
+        {
+          const double X_i = get_ele(*(x), 0);
+          const double f_i = get_ele(*(xDot), 0);
+
+          if (stageNumber == 0) {
+            TEST_FLOATING_EQUALITY(X_i,      0.5, relTol); 
+            TEST_FLOATING_EQUALITY(time,     1.0, relTol);   
+            TEST_FLOATING_EQUALITY(f_i,     -0.5, relTol);   
+          } else if (stageNumber == 1) {
+            TEST_FLOATING_EQUALITY(X_i,     0.75, relTol);
+            TEST_FLOATING_EQUALITY(f_i,    -0.75, relTol);   
+            TEST_ASSERT(std::abs(time) < relTol);
+          } else {
+            TEUCHOS_TEST_FOR_EXCEPT( !(-1 < stageNumber && stageNumber < 2));
+          }
+
+          break;
+        }
+      case StepperRKAppAction<double>::END_STEP:
+        {
+          const double x_1 = get_ele(*(x), 0);
+          time = workingState->getTime();
+          TEST_FLOATING_EQUALITY(x_1,  3.0/8.0, relTol);
+          TEST_FLOATING_EQUALITY(time,     1.0, relTol);
+          TEST_FLOATING_EQUALITY(dt,       1.0, relTol);
+          TEST_COMPARE(stageNumber, ==, -1);
+
+          if (stepper->getUseEmbedded() == true) {
+            TEST_FLOATING_EQUALITY(workingState->getTolRel(), 1.0, relTol);
+            TEST_ASSERT(std::abs(workingState->getTolAbs()) < relTol);
+            // e = 0 from doxygen above.
+            TEST_ASSERT(std::abs(workingState->getErrorRel()) < relTol);
+          }
+
+        } 
+
+    }
+  }
+
+private:
+
+  Teuchos::FancyOStream & out;
+  bool & success;
+};
+
+
+// ************************************************************
+// ************************************************************
+TEUCHOS_UNIT_TEST(DIRK, Modifier)
+{
+  auto stepper = rcp(new Tempus::StepperSDIRK_21Pair<double>());
+  Teuchos::RCP<const Thyra::ModelEvaluator<double> >
+    model = rcp(new Tempus_Test::DahlquistTestModel<double>());
+
+  //auto modifier = rcp(new StepperRKModifierBogackiShampineTest(out, success));
+
+  stepper->setModel(model);
+  //stepper->setAppAction(modifier);
+  stepper->setICConsistency("Consistent");
+  stepper->setUseFSAL(false);
+  stepper->initialize();
+
+  // Create a SolutionHistory.
+  auto solutionHistory = Tempus::createSolutionHistoryME(model);
+
+  // Take one time step.
+  stepper->setInitialConditions(solutionHistory);
+  solutionHistory->initWorkingState();
+  double dt = 1.0;
+  solutionHistory->getWorkingState()->setTimeStep(dt);
+  solutionHistory->getWorkingState()->setTime(dt);
+  stepper->takeStep(solutionHistory);           // Primary testing occurs in
+                                                // modifierX during takeStep().
+  // Test stepper properties.
+  TEUCHOS_ASSERT(stepper->getOrder() == 2);
+}
+
+} // namespace Tempus_Test
```

Comment:
It looks like this is focused on SDIRK21.  Should it be in Tempus_UnitTest_SDIRK_21Pair.cpp?

Reply:
I was hoping to add EDIRK Trapezoid to this test as well, per the comment in the issue. What do you think?

Reply:
I guess I would put that in Tempus_UnitTest_EDIRK_TrapezoidalRule.cpp.  I was thinking we should group by Stepper.

Reply:
I've just pushed the ESDIRK_Trapezoid test section

Reply:
Do you want to put these in this file or in Tempus_UnitTest_SDIRK_21Pair.cpp and Tempus_UnitTest_EDIRK_TrapezoidalRule.cpp?

Reply:
Done